frontend/src/views/Collection/components/DatasetTab/index.tsx

```diff
@@ -49,9 +49,10 @@ const DatasetTab: FC<Props> = ({
   const isDatasetPresent =
     datasets?.length > 0 || Object.keys(uploadedFiles).length > 0;
 
-  const invalidateCollectionQuery = () => {
-    queryClient.invalidateQueries([USE_COLLECTION, collectionId]);
-  };
+  const invalidateCollectionQuery = () =>
+    useCallback(() => {
+      queryClient.invalidateQueries([USE_COLLECTION, collectionId]);
+    }, [collectionId]);
```

Comment:
We can drop the outer function, otherwise we'll be creating a new function every time `DatasetTab()` function is called still 😄  Thank you!

```ts
const invalidateCollectionQuery = useCallback(() => {
  queryClient.invalidateQueries([USE_COLLECTION, collectionId]);
}, [collectionId]);
```

Reply:
I don't know how I did this. This was accidental. Great reason for code review.

Reply:
Totally! We all look out for each other 🎉 😄